icu4c/source/i18n/unitsdata.cpp

```diff
@@ -0,0 +1,194 @@
+        ResourceTable conversionRateTable = value.getTable(status);
+        if (U_FAILURE(status)) return;
+
+        // Collect base unit, factor and offset from the resource.
+        int32_t lenSource = uprv_strlen(source);
+        const UChar *baseUnit = NULL, *factor = NULL, *offset = NULL;
+        int32_t lenBaseUnit, lenFactor, lenOffset;
+        const char *key;
+        for (int32_t i = 0; conversionRateTable.getKeyAndValue(i, key, value); ++i) {
+            if (uprv_strcmp(key, "target") == 0) {
+                baseUnit = value.getString(lenBaseUnit, status);
+            } else if (uprv_strcmp(key, "factor") == 0) {
+                factor = value.getString(lenFactor, status);
+            } else if (uprv_strcmp(key, "offset") == 0) {
+                offset = value.getString(lenOffset, status);
+            }
+        }
+        if (U_FAILURE(status)) return;
+        if (baseUnit == NULL || factor == NULL) {
+            // We could not find a usable conversion rate.
+            status = U_MISSING_RESOURCE_ERROR;
+            return;
+        }
+
+        // Check if we already have the conversion rate in question.
+        //
+        // TODO(review): We could do this skip-check *before* we fetch
+        // baseUnit/factor/offset based only on the source unit, but only if
+        // we're certain we'll never get two different baseUnits for a given
+        // source. This should be the case, since convertUnit entries in CLDR's
+        // units.xml should all point at a defined base unit for the unit
+        // category. I should make this code more efficient after
+        // double-checking we're fine with relying on such a detail from the
+        // CLDR spec?
+        fLastBaseUnit.clear();
+        fLastBaseUnit.appendInvariantChars(baseUnit, lenBaseUnit, status);
+        if (U_FAILURE(status)) return;
+        for (int32_t i = 0, len = outVector.length(); i < len; i++) {
+            if (strcmp(outVector[i]->sourceUnit.data(), source) == 0 &&
+                strcmp(outVector[i]->baseUnit.data(), fLastBaseUnit.data()) == 0) {
+                return;
+            }
+        }
+
+        // We don't have this ConversionRateInfo yet: add it.
+        ConversionRateInfo *cr = outVector.emplaceBack();
+        if (!cr) {
+            status = U_MEMORY_ALLOCATION_ERROR;
+            return;
+        } else {
+            cr->sourceUnit.append(source, lenSource, status);
+            cr->baseUnit.append(fLastBaseUnit.data(), fLastBaseUnit.length(), status);
+            cr->factor.appendInvariantChars(factor, lenFactor, status);
+            if (offset != NULL) cr->offset.appendInvariantChars(offset, lenOffset, status);
+        }
+    }
+
+    /**
+     * Returns the MeasureUnit that was the conversion base unit of the most
+     * recent call to put() - typically meaning the most recent call to
+     * ures_getAllItemsWithFallback().
+     */
+    MeasureUnit getLastBaseUnit(UErrorCode &status) {
```

Comment:
Why do you need this?

Reply:
That was my code - it was a more hacky work-around within https://github.com/sffc/icu/pull/31 before the [Rip out compound base unit calculation](https://github.com/sffc/icu/pull/31/commits/cd32130c7f49b28271e314848161d8b3717c51b3) commit.

I'll fix it in my pull request, FWIW. Then I need to coordinate with @younies : I can make a pull-request into your pull-request, or I can simply push the commit into your branch, whichever you prefer.

Reply:
This is related to Hugo pull request: #31

Reply:
This is gone now, marking as resolved.